unified-runtime/source/adapters/level_zero/program.cpp

```diff
@@ -406,7 +446,14 @@ ur_result_t urProgramLinkExp(
 
     ZeStruct<ze_module_desc_t> ZeModuleDesc;
     ZeModuleDesc.pNext = &ZeExtModuleDesc;
-    ZeModuleDesc.format = ZE_MODULE_FORMAT_IL_SPIRV;
+    switch (CommonCodeFormat) {
+    case ur_program_handle_t_::CodeFormat::SPIRV:
+      ZeModuleDesc.format = ZE_MODULE_FORMAT_IL_SPIRV;
+      break;
+    default:
+      assert(false && "Unexpected code format");
```

Comment:
should we assert here?
Maybe we should just log something instead:
```
UR_LOG(ERR, "urProgramLinkExp: Unexpected code format");
```

Reply:
If we get here then `program->getCodeFormat()` returned something we did not expect. That should never happen as invalid formats are supposed to be rejected at `urCreateProgramWithIL` time. Therefore this is a programming error in UR and IMO those should be asserts.

Reply:
Makes sense. Thanks. Use `ur::unreachable()` in that case.